doc/v5.0/man3/MPI_File_write_ordered_end.3
.\" Automatically generated by Pandoc 2.14.2
.\"
.TH "MPI_File_write_ordered_end" "3" "" "2021-08-30" "Open MPI"
.hy
.SH Name
.PP
\f[C]MPI_File_write_ordered_end\f[R] - Writes a file at a location
specified by a shared file pointer; ending part of a split collective
routine (blocking).
.SH Syntax
.SS C Syntax
.IP
.nf
\f[C]
#include <mpi.h>

int MPI_File_write_ordered_end(MPI_File fh, const void *buf,
    MPI_Status *status)
\f[R]
.fi
.SS Fortran Syntax
.IP
.nf
\f[C]
USE MPI
! or the older form: INCLUDE \[aq]mpif.h\[aq]

MPI_FILE_WRITE_ORDERED_END(FH, BUF, STATUS, IERROR)
    <type>  BUF(*)
    INTEGER FH, STATUS(MPI_STATUS_SIZE), IERROR
\f[R]
.fi
.SS Fortran 2008 Syntax
.IP
.nf
\f[C]
USE mpi_f08

MPI_File_write_ordered_end(fh, buf, status, ierror)
    TYPE(MPI_File), INTENT(IN) :: fh
    TYPE(*), DIMENSION(..), INTENT(IN), ASYNCHRONOUS :: buf
    TYPE(MPI_Status) :: status
    INTEGER, OPTIONAL, INTENT(OUT) :: ierror
\f[R]
.fi
.SH Input/Output Parameter
.IP \[bu] 2
\f[C]fh\f[R] : File handle (handle).
.SH Input Parameter
.IP \[bu] 2
\f[C]buf\f[R] : Initial address of buffer (choice).
.SH Output Parameters
.IP \[bu] 2
\f[C]status\f[R] : Status object (status).
.IP \[bu] 2
\f[C]IERROR\f[R] : Fortran only: Error status (integer).
.SH Description
.PP
\f[C]MPI_File_write_ordered_end\f[R] is the ending part of a split
collective routine that must be called by all processes in the
communicator group associated with the file handle \f[C]fh\f[R].
\f[C]MPI_File_write_ordered_end\f[R] returns the number of elements
written into the file associated with \f[C]fh\f[R] in \f[C]status\f[R].
.SH Notes
.PP
All the nonblocking collective routines for data access are
\[lq]split\[rq] into two routines, each with _begin or _end as a suffix.
These split collective routines are subject to the semantic rules
described in Section 9.4.5 of the MPI-2 standard.
.SH Errors
.PP
Almost all MPI routines return an error value; C routines as the value
of the function and Fortran routines in the last argument.
.PP
Before the error value is returned, the current MPI error handler is
called.
For MPI I/O function errors, the default error handler is set to
\f[C]MPI_ERRORS_RETURN\f[R].
The error handler may be changed with \f[C]MPI_File_set_errhandler\f[R];
the predefined error handler \f[C]MPI_ERRORS_ARE_FATAL\f[R] may be used
to make I/O errors fatal.
Note that MPI does not guarantee that an MPI program can continue past
an error.


doc/v1.4/man3/MPI_File_write_ordered_end.3.php
<?php
$topdir = "../../..";
$title = "MPI_File_write_ordered_end(3) man page (version 1.4.5)";
$meta_desc = "Open MPI v1.4.5 man page: MPI_FILE_WRITE_ORDERED_END(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
  <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_File_write_ordered_end</b> - Writes a file at a location specified
by a shared file pointer; ending part of a split collective routine (blocking).

<p>
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>
<br>
<pre>C Syntax
    #include &lt;mpi.h&gt;
    int MPI_File_write_ordered_end(MPI_File fh, void *buf,
    <tt> </tt>&nbsp;<tt> </tt>&nbsp;      MPI_Status *status)
</pre>
<h2><a name='sect2' href='#toc2'>Fortran Syntax</a></h2>
<br>
<pre>    INCLUDE &rsquo;mpif.h&rsquo;
    MPI_FILE_WRITE_ORDERED_END(FH, BUF, STATUS, IERROR)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;<tt> </tt>&nbsp;<tt> </tt>&nbsp;&lt;type&gt;<tt> </tt>&nbsp;<tt> </tt>&nbsp;<tt> </tt>&nbsp;<tt> </tt>&nbsp;BUF(*)
        <tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER<tt> </tt>&nbsp;<tt> </tt>&nbsp;<tt> </tt>&nbsp;<tt> </tt>&nbsp;FH, STATUS(MPI_STATUS_SIZE), IERROR
</pre>
<h2><a name='sect3' href='#toc3'>C++ Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
void MPI::File::Write_ordered_end(const void* buf, MPI::Status&amp; status)
void MPI::File::Write_ordered_end(const void* buf)
</pre>
<h2><a name='sect4' href='#toc4'>Input/Output Parameter</a></h2>

<dl>

<dt>fh     </dt>
<dd>File handle (handle).
<p> </dd>
</dl>

<h2><a name='sect5' href='#toc5'>Input Parameter</a></h2>

<dl>

<dt>buf </dt>
<dd>Initial
address of buffer (choice).
<p> </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Output Parameters</a></h2>

<dl>

<dt>status </dt>
<dd>Status object (status).
 </dd>

<dt>IERROR </dt>
<dd>Fortran only: Error status (integer).
<p> </dd>
</dl>

<h2><a name='sect7' href='#toc7'>Description</a></h2>
MPI_File_write_ordered_end
is the ending part of a split collective routine that must be called by
all processes in the communicator group associated with the file handle
 <i>fh.</i> MPI_File_write_ordered_end returns the number of elements written into
the file associated with  <i>fh</i> in <i>status.</i>
<p>
<h2><a name='sect8' href='#toc8'>Notes</a></h2>
All the nonblocking collective
routines for data access are "split" into two routines, each with _begin
or _end as a suffix. These split collective routines are subject to the
semantic rules described in Section 9.4.5 of the MPI-2 standard.
<p>
<h2><a name='sect9' href='#toc9'>Errors</a></h2>
Almost
all MPI routines return an error value; C routines as the value of the
function and Fortran routines in the last argument. C++ functions do not
return errors. If the default error handler is set to MPI::ERRORS_THROW_EXCEPTIONS,
then on error the C++ exception mechanism will be used to throw an MPI:Exception
object. <p>
Before the error value is returned, the current MPI error handler
is called. For MPI I/O function errors, the default error handler is set
to MPI_ERRORS_RETURN. The error handler may be changed with <a href="../man3/MPI_File_set_errhandler.3.php">MPI_File_set_errhandler</a>;
the predefined error handler MPI_ERRORS_ARE_FATAL may be used to make I/O
errors fatal. Note that MPI does not guarantee that an MPI program can continue
past an error.
<p> <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>Fortran Syntax</a></li>
<li><a name='toc3' href='#sect3'>C++ Syntax</a></li>
<li><a name='toc4' href='#sect4'>Input/Output Parameter</a></li>
<li><a name='toc5' href='#sect5'>Input Parameter</a></li>
<li><a name='toc6' href='#sect6'>Output Parameters</a></li>
<li><a name='toc7' href='#sect7'>Description</a></li>
<li><a name='toc8' href='#sect8'>Notes</a></li>
<li><a name='toc9' href='#sect9'>Errors</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
